### PR TITLE
PRMT-3413 E2E testing - Unexpected Interaction ID and NHS number

### PR DESCRIPTION
**Pull request by joefong-nhs**
- Move the tests from previous branch (PRMT-3413) to this branch 
- Modify the tests to adapt to the configurations after major refactoring
- Add HealthCheckService and EhrOutServiceProperties in order to call the health check endpoints of each services
- Remove some unused imports

### CODE REVIEW

File: src/main/java/uk/nhs/prm/e2etests/property/EhrOutServiceProperties.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.prm.e2etests.property;
+
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.nhs.prm.e2etests.service.SsmService;
+
+@Getter
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/property/EhrOutServiceProperties.java:9]: I would perhaps lean towards manual getters for the fields required!

File: src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package uk.nhs.prm.e2etests.service;
+
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java:5]: Can we opt for single line imports over wildcard please!

File: src/test/java/uk/nhs/prm/e2etests/test/RepositoryE2ETest.java [hunk 3]
```diff
@@ -1,10 +1,7 @@
 package uk.nhs.prm.e2etests.test;
 
 import lombok.extern.log4j.Log4j2;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.*;
```
Comment (martin-nhs) [inline src/test/java/uk/nhs/prm/e2etests/test/RepositoryE2ETest.java:4]: Can we opt for single line imports over wildcard please!

File: src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java (hunk 2 above)
Comment (joefong-nhs) [reply src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java]: Agree! PR addressed :)

File: src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            throw new ServiceException(getClass().getName(), exception.getMessage());
+        }
+    }
+
+    private HttpHeaders createHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return headers;
+    }
+
```
[the remark this answers is not in the document]
Comment (joefong-nhs) [reply src/main/java/uk/nhs/prm/e2etests/service/HealthCheckService.java]: PR addressed! Thanks

File: src/test/java/uk/nhs/prm/e2etests/test/RepositoryE2ETest.java (hunk 3 above)
Comment (joefong-nhs) [reply src/test/java/uk/nhs/prm/e2etests/test/RepositoryE2ETest.java]: PR addressed! Thanks

**Review by martin-nhs — APPROVED** [review]

File: src/main/java/uk/nhs/prm/e2etests/queue/AbstractMessageQueue.java [hunk 5]
```diff
@@ -74,6 +77,31 @@ public List<SqsMessage> getAllMessageContaining(String substring, int expectedNu
         return allMessages;
     }
 
+
+
+
+    public boolean verifyNoMessageContaining(String substring, int secondsToPoll) {
+        // Queue the queue repeatedly for {secondsToPoll}, and return true only if a message with given substring NEVER appeared throughout the period.
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/e2etests/queue/AbstractMessageQueue.java:84]:
```suggestion
        // Poll the queue repeatedly for {secondsToPoll}, and return true only if a message with given substring NEVER appeared throughout the period.
```

**Pull request merged**